routes/settings.js

```diff
@@ -39,7 +39,12 @@ router.post('/template', async (req, res) => {
 
 router.post('/', async (req, res) => {
     const formData = req.body;
-    formData['ghosler.url'] = req.protocol + '://' + req.hostname;
+
+    let fullUrl = `${req.protocol}://${req.get('Host')}${req.originalUrl.split("/settings")[0]}`;
+    if (req.get('Referer')) {
+        fullUrl = req.get('Referer').split("/settings")[0];
+    }
```

Comment:
Yup `req.get('Host')` is supposed to be the modern way of accessing it.

As far as the difference, the split on the end theoretically grabs the subdirectory uri.

It may not ever actually be useful though. The theory was that if someone happened to be passing the full uri to Ghosler it would get the subdirectory. But the way I've deployed I didn't ever really hit it and instead got the functionality from the Referer. Figured it didn't hurt to have in place though 🤷‍♂️

Reply:
Oh in case you were asking about `req.originalUrl` it gives access to the full path and query. So what it is doing is taking the path and slicing off the /settings and anything that possibly could be to the right of it.

Maybe it would be better to construct it as a URL object first and then grab what we need from it. Might be a little safer.

Reply:
Implemented URL object parsing for safety

Reply:
My point is that the url will anyway include `/settings` to the right of `host`.
Why use `originalUrl` in the first place if we are anyway splitting it later 😅? Or am I missing something?

Example : 
Your implementations returns - `http://localhost:2369/` , but the same is returned via -
```js
formData['ghosler.url'] = `${req.protocol}://${req.get('host')}/`;
// formData['ghosler.url'] = new URL(`${req.protocol}://${req.get('host')}`).toString();
```

Reply:
The original think was that if someone somehow served it at `http://localhost:2369/ghosler/path/whatever/settings` it would capture the ghosler.url as `http://localhost:2369/ghosler/path/whatever` but in reality as i said initially i don't think it would ever actually capture it  there and instead be done by the Referer header. Just left it in just in case.

Reply:
Aaahh I see what you mean now!! My bad for the confusion.

Thanks for the explanation!
Just confirming again -
1. Assuming the app is installed at - `https://xyz.com/newsletters`

2. So the settings would be - `https://xyz.com/newsletters/settings`

3. This change would use `https://xyz.com/newsletters` as the base domain to be saved in the configuration

Right??

Reply:
Yup that’s how it should work in theory.

in practice though as most users would have this behind a reverse proxy that is pointing at `https://xyz.com` it doesn’t get the right URI here and instead gets it from the Referer. But it won’t hurt anyone who is hosting it the normal way.

I think there’s probably a way to define the hostname and path in separate config variables and dynamically define the routes so that it would function properly and make for a slightly less error prone reverse proxy config. But I haven’t fully thought through the implications for the route definitions if we did so.

Reply:
Going ahead with this as is and may revisit in future for better out the box experience